### ISSUE TITLE
Provide an option to make "you provided 'undefined' where a stream was expected" an error instead of warning

### ISSUE THREAD

**Issue by biltongza** [ReactiveX/rxjs#5876]
I have just found a sneaky bug in our code where someone was using `combineLatest()` with an observable that was not yet initialised. Is there a way to turn this into a fatal error that stops the stream instead of a warning that just carries on working anyway? In cases like this I would like to be able to fail fast so we are aware there is a mistake sooner rather than later.

Stackblitz: https://stackblitz.com/edit/rxjs-ltdymt

```
import { of, combineLatest } from 'rxjs'; 
import { map } from 'rxjs/operators';


const source = of('World').pipe(
  map(x => `Hello ${x}!`)
);

combineLatest([source, undefined]).subscribe(([hello, undef]) => console.log(hello, undef));
```

Result in console:
>TypeError: You provided 'undefined' where a stream was expected. You can provide an Observable, Promise, Array, or Iterable.

It is marked as a warning in the console:
![image](https://user-images.githubusercontent.com/2845704/98232435-52ba5300-1f66-11eb-8fe6-0d32d84711c8.png)

I see in my example now it is actually stopping the stream as expected. I must have made a mistake when I thought it doesn't stop the stream.

**Comment by cartant** [comment]
> ... stops the stream instead of a warning that just carries on working anyway?

What does this mean? AFAICT, it's an [error that should stop the stream](https://github.com/ReactiveX/rxjs/blob/1e1e022ed9fdc4a4bfed63cf151887031ac57398/src/internal/util/subscribeTo.ts#L24-L27). A snippet that serves as a minimal reproduction of the problem would be helpful. There is a reason maintainers ask for the issue template to be filled out.

**Comment by biltongza** [comment]
Updated OP, I see in my example it is actually stopping the stream. It would be nice if it were logged as an error instead of a warning in the console though.

**Comment by cartant** [comment]
Closing this because it's working as intended. If there is no error handler specified in the `subscribe` call, an error is thrown, but RxJS throws errors that cannot be reported to an observer *asynchronously* to avoid producer interference.

**Comment by biltongza** [comment]
Is the asynchronous throw the reason it is a warning instead of an error in the console? I'm really just looking for more visibility here.

**Comment by cartant** [comment]
Ah, I see what you mean. However, this is still by design/necessity, as this was a bug that I fixed - with synchronous-on-subscribe errors that couldn't be reported - *after* version 6 was released. Prior to the fix those kinds of errors were swallowed. It was deemed that throwing would have been a breaking change. Logging at least made the errors visible.

This can stay closed because it's addressed in version 7: https://ncjamieson.com/stopped-notifications/
